commands/operator-sdk/cmd/build.go

```diff
@@ -168,10 +153,14 @@ func buildFunc(cmd *cobra.Command, args []string) {
 
 	log.Infof("Building Docker image %s", baseImageName)
 
-	dbcmd := exec.Command("docker", "build", ".", "-f", "build/Dockerfile", "-t", baseImageName)
-	dbcmd.Stdout = os.Stdout
-	dbcmd.Stderr = os.Stderr
-	err := dbcmd.Run()
+	buildDockerfile := filepath.Join(scaffold.BuildDir, scaffold.DockerfileFile)
+	buildDockerfile = makeDockerfileIfMultistage(buildDockerfile, false)
+	if projutil.IsOperatorGo() && !projutil.IsDockerfileMultistage(buildDockerfile) {
```

Comment:
Nit: Can we add a comment of explanation as it's a lot of conditions here, so we know in the future?

Reply:
Sure. I'll add comments to changes with moderate complexity like this one.